plugin/veracode_static_scan.go

```diff
@@ -0,0 +1,299 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func runVeracodeStaticScanPlugin(ctx context.Context, args Args) error {
+	logrus.Infof("🟢 Starting Veracode UploadAndScan")
+
+	if err := ValidateInputs(args); err != nil {
+		return fmt.Errorf("input validation failed: %v", err)
+	}
+
+	finalFileList, err := resolveUploadFileList(args.UploadIncludesPattern, args.UploadExcludesPattern)
+	if err != nil {
+		return fmt.Errorf("failed to resolve upload file list: %w", err)
+	}
+
+	cmdArgs := buildVeracodeCommandArgs(args, finalFileList)
+	maskedArgs := maskSensitiveArgs(cmdArgs)
+	logrus.Infof("➡️  Executing: java %s", strings.Join(maskedArgs, " "))
+
+	cmd := exec.CommandContext(ctx, "java", cmdArgs...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	err = runJavaCommandWithTimeout(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	if args.WaitForScan {
+		return waitForScanCompletion(ctx, args)
+	}
+
+	return nil
+}
+
+func maskSensitiveArgs(args []string) []string {
+	masked := make([]string, len(args))
+	copy(masked, args)
+	for i := 0; i < len(masked)-1; i++ {
+		if masked[i] == "-vkey" || masked[i] == "-vid" || masked[i] == "-ppassword" {
+			masked[i+1] = "****"
+		}
+	}
+	return masked
+}
+
+func buildVeracodeCommandArgs(args Args, fileList string) []string {
+	cmdArgs := []string{
+		"-jar", "/opt/veracode/api-wrapper.jar",
+		"-action", "UploadAndScan",
+		"-appname", args.AppName,
+		"-filepath", fileList,
+		"-version", args.Version,
+		"-vid", args.VID,
+		"-vkey", args.VKey,
+	}
+
+	optional := []struct {
+		flag  string
+		value string
+	}{
+		{"-criticality", args.Criticality},
+		{"-sandboxname", args.SandboxName},
+		{"-teams", args.Teams},
+		{"-pattern", args.FileNamePattern},
+		{"-replacement", args.ReplacementPattern},
+		{"-exclude", args.ScanExcludesPattern},
+		{"-include", args.ScanIncludesPattern},
+		{"-phost", args.PHost},
+		{"-pport", args.PPort},
+		{"-puser", args.PUser},
+		{"-ppassword", args.PPassword},
+	}
+
+	for _, opt := range optional {
+		if opt.value != "" {
+			cmdArgs = append(cmdArgs, opt.flag, opt.value)
+		}
+	}
+
+	flags := []struct {
+		flag string
+		cond bool
+	}{
+		{"-createprofile", args.CreateProfile},
+		{"-createsandbox", args.CreateSandbox},
+		{"-debug", args.Debug},
+		{"-scanallnonfataltoplevelmodules", args.ScanAllNonFatalTopLevel},
+		{"-includenewmodules", args.IncludeNewModules},
+		{"-deleteincompletescan", args.DeleteIncompleteScan},
+	}
+
+	for _, f := range flags {
+		if f.cond {
+			cmdArgs = append(cmdArgs, f.flag, "true")
+		}
+	}
+
+	return cmdArgs
+}
+
+func runJavaCommandWithTimeout(cmd *exec.Cmd, args Args) error {
+	done := make(chan error, 1)
+	go func() {
+		done <- cmd.Run()
+	}()
+
+	if args.Timeout > 0 {
+		select {
+		case err := <-done:
+			return handleResult(err, args)
+		case <-time.After(time.Duration(args.Timeout) * time.Minute):
```

Comment:
when a timeout occurs the process isn't being explicitly killed which could lead to zombie processes. consider adding `cmd.Process.Kill()` when timing out and properly handling context cancellation